processors/filterprocessor/filter_processor_test.go

```diff
@@ -0,0 +1,272 @@
+package filterprocessor
+
+import (
+	"context"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/indexer/conduit"
+	"github.com/algorand/indexer/data"
+	"github.com/algorand/indexer/plugins"
+	"github.com/algorand/indexer/processors"
+)
+
+// TestFilterProcessor_Init_Multi tests initialization of the filter processor with the "all" and "any" filter types
+func TestFilterProcessor_Init_Multi(t *testing.T) {
+
+	sampleAddr1 := basics.Address{1}
+	sampleAddr2 := basics.Address{2}
+	sampleAddr3 := basics.Address{3}
+
+	sampleCfgStr := `---
+filters:
+  - any: 
+    - tag: SignedTxnWithAD.SignedTxn.AuthAddr
+      expression-type: exact
+      expression: "` + sampleAddr1.String() + `"
+    - tag: SignedTxnWithAD.SignedTxn.Txn.AssetTransferTxnFields.AssetSender
+      expression-type: regex
+      expression: "` + sampleAddr3.String() + `"
+  - all:
+    - tag: SignedTxnWithAD.SignedTxn.Txn.PaymentTxnFields.Receiver
+      expression-type: regex 
+      expression: "` + sampleAddr2.String() + `"
+    - tag: SignedTxnWithAD.SignedTxn.Txn.Header.Sender
+      expression-type: exact
+      expression: "` + sampleAddr2.String() + `"
+  - any: 
+    - tag: SignedTxnWithAD.SignedTxn.Txn.AssetTransferTxnFields.AssetCloseTo
+      expression-type: exact
+      expression: "` + sampleAddr2.String() + `"
+    - tag: SignedTxnWithAD.SignedTxn.Txn.AssetTransferTxnFields.AssetReceiver
+      expression-type: regex
+      expression: "` + sampleAddr2.String() + `"
```

Comment:
what happens if all and any have overlapping tags?

Reply:
The filters are evaluated in a top down order.  Meaning that the first filter that gets encountered gets executed.

Reply:
That's a feature isn't it? i.e. if you want to match two different accounts:
```
- any:
    - tag:  SignedTxnWithAD.SignedTxn.Txn.Sender
      expression-type: const
      expression: "AAAAAAA..."
    - tag:  SignedTxnWithAD.SignedTxn.Txn.Sender
      expression-type: const
      expression: "BBBBBBB..."
```